### ISSUE TITLE
Microphone and Camera buttons not working

### ISSUE THREAD

**Issue by tevincent** [jitsi/jitsi-meet#14794]
### What happened?

Hello,

The first time I'm joining a call, just after installing the app, camera and microphone buttons are not working. Is that a known issue ? Or did I forget to implement something ?

The way I'm able to reproduce it is:
* Install the app
* Join a call
* Accept the microphone permission
* Then all buttons are not working

I have attached a video to show the problem.

Thanks !

### Platform

- [ ] Chrome (or Chromium based)
- [ ] Firefox
- [ ] Safari
- [ ] Other desktop browser
- [ ] Android browser
- [ ] iOS browser
- [ ] Electron app
- [X] Android mobile app
- [ ] iOS mobile app
- [ ] Custom app using a mobile SDK

### Browser / app / sdk version

9.2.2

### Reproducibility

- [ ] The problem is reproducible on meet.jit.si

**Comment by tevincent** [comment]
https://github.com/jitsi/jitsi-meet/assets/149579879/f74f79c3-ae09-41db-8149-2f3e66e7c2a0

**Comment by saghul** [comment]
Hey there! How did you integrate the SDK into your app? Did you use our Activity?

**Comment by saghul** [comment]
That looks all good! Anything in `adb logcat` ? Do other toolbar buttons work?

**Comment by tevincent** [comment]
All the other buttons are working. Here is our repo:

https://github.com/Infomaniak/android-infomaniak-meet

The only difference is that I'm trying to update to the latest version of the SDK, which is 9.2.2.

P.S: Thanks for the prompt response!

**Comment by saghul** [comment]
Very odd, I don't see anything obvious on the surface :-/ I wonder if it has to do with some support library version mismatch...

**Comment by saghul** [comment]
I don't know 100%, it could be, I guess. It doesn't make much sense TBH.

**Comment by tevincent** [comment]
I just tried to update everything to the latest version, same issue.

**Comment by saghul** [comment]
Can you please share the entire logcat output?

**Comment by tevincent** [comment]
Here it is.
[log_meet.txt](https://github.com/user-attachments/files/15545901/log_meet.txt)

**Comment by saghul** [comment]
Can yoy adding `.setFeatureFlag("resolution", 360)` on the conference options to see if it makes a difference?

**Comment by tevincent** [comment]
Just tried, same.

**Comment by saghul** [comment]
I assume this happens on several devices and not just this one, right?

**Comment by saghul** [comment]
Does the official Jitsi Meet app exhibit the same issue?

**Comment by tevincent** [comment]
Nope. Everything is working fine on the Jitsi Meet App.

**Comment by tevincent** [comment]
But one thing to note, it's working fine with the SDK 8.6.0.

**Comment by tevincent** [comment]
One more info. I tried all version from 8.6.6 to 9.2.2 and the problem is only on the version 9.2.2. Hope it helps !

**Comment by saghul** [comment]
@Calinteodor does this ring a bell?

**Comment by tevincent** [comment]
Another info, the problem is only on device above API 32. Everything is working fine from API32 and downward.

**Comment by tevincent** [comment]
It seems to work. One thing that might be interested. If I do this:
```
...
.setFeatureFlag("prejoinpage.enabled", true)
...
```
I can enable microphone and camera. But if this is false, I cannot.

I cannot really do the same thing in the sample since I have to login on Jitsi meet etc.

**Comment by Calinteodor** [comment]
Yep. But after enabling video and audio in Prejoin you can do the same inside the conference, right?

**Comment by tevincent** [comment]
Nope. You can see I try to click on the camera and icon and nothing happens.

**Comment by Calinteodor** [comment]
If you have prejoin page enabled on your app, I am asking.

**Comment by tevincent** [comment]
Oh ok my bad. Yes if I have the prejoin screen and I enable the microphone and camera, in the conference screen, everything works as expected.

**Comment by Calinteodor** [comment]
Ok. I took another look and I tested using the sample app. 
Disabled prejoin page, started muted with both video and audio.

https://github.com/jitsi/jitsi-meet/assets/26413496/ed66b45f-faca-4731-94bb-39a8a8dafa7f

Everything seems to be in order. Can you check the sample again? First, just the way it is.

**Comment by tevincent** [comment]
Do you want me to push a branch with the bug ? I can reproduce easily.

**Comment by Calinteodor** [comment]
Let me try a fresh start.

**Comment by Calinteodor** [comment]
> Let me try again. But did you try the first time you launch the app ? Because I can see it didn't ask for any permission in your video. Because in my case, if it's a fresh install -> launch app -> call, I have the issue but then it's working.


Let me try a fresh start.

**Comment by Calinteodor** [comment]
> > Let me try again. But did you try the first time you launch the app ? Because I can see it didn't ask for any permission in your video. Because in my case, if it's a fresh install -> launch app -> call, I have the issue but then it's working.
> 
> Let me try a fresh start.

Sorry, but I still don't repro the bug. Microphone mutes/unmutes as expected. I get all the permission notifications and allow them. Nothing unexpected. Can you send your server url so I can test with that, too?

**Comment by tevincent** [comment]
Here it is: "https://kmeet.infomaniak.com/"

Let me know if you want me to create a branch with the modification to reproduce the bug.

**Comment by Calinteodor** [comment]
> Here it is: https://kmeet.infomaniak.com/
> 
> Let me know if you want me to create a branch with the modification to reproduce the bug.

Let's do that and I will check your code changes on the sample app.

**Comment by tevincent** [comment]
Here it is: https://github.com/jitsi/jitsi-meet-sdk-samples/pull/236

**Comment by Calinteodor** [comment]
I tested using meet.jit.si server and everything works as expected. 
When I test using your server, the problem is reproducible. 
I don't know what setting on your side could cause this issue.

**Comment by saghul** [comment]
What versions of the server components are you running @tevincent ?

**Comment by tevincent** [comment]
That's crazy. How the configuration of the server could impact the behavior of the microphone and camera button ?

**Comment by tevincent** [comment]
> What versions of the server components are you running @tevincent ?

Let me get back to you when I get the answer.

**Comment by saghul** [comment]
https://github.com/jitsi/jitsi-meet/releases/tag/stable%2Fjitsi-meet_8615

So that is 1 year old.

It _should_ work, but we might have messed up something inadvertently.

Let's try something to validate that a server update would "solve" it: please join a meeting in alpha.jitsi.net with your app. Does it work?

Note: alpha uses the latest of everything and meetings are destroyed after 5 minutes.

**Comment by tevincent** [comment]
Just tried with https://alpha.jitsi.net. The microphone button seems to be working but not the camera button.


https://github.com/jitsi/jitsi-meet/assets/149579879/394d1575-1d10-446b-a023-36b1d2832f98

**Comment by saghul** [comment]
@Calinteodor

**Comment by Calinteodor** [comment]
This is very weird. I took another look using the sample android kotlin app with prejoin page disabled and server set to https://alpha.jitsi.net/ and all buttons work as expected.

**Comment by viking2917** [comment]
Possibly related issue, which JITSI folks say will be fixed in next few weeks: https://github.com/jitsi/jitsi-meet-sdk-samples/issues/239#issuecomment-2252531295

**Comment by Calinteodor** [comment]
Jitsi Meet SDK Android and iOS 10.0.0 have been released. You can check the full release log right here
https://github.com/jitsi/jitsi-meet-release-notes/blob/master/CHANGELOG-MOBILE-SDKS.md#1000-2024-08-07

**Comment by tevincent** [comment]
Same issue. I tried with this server url: `https://alpha.jitsi.net/`

**Comment by Calinteodor** [comment]
> Same issue with 10.0.0. I tried with our own server and this one: `https://alpha.jitsi.net/`

Tested myself one more time using the sample app with 10.0.0 and https://alpha.jitsi.net/.
Receiving all permission notifications and audio/video button works as expected,

**Comment by viking2917** [comment]
I have been testing with the 10.0.0 SDK built into my app, and the audio/video mute buttons are working as expected. The problem was always intermittent for me, so I am not sure this is definitive, but it seems to be working.

**Comment by tevincent** [comment]
After discussing with my team, we've set `prejoinpage.enabled` and `welcomepage.enabled` to `true`. As a result, we haven't encountered any issues. Since no one else is reproducing this bug, perhaps we can close this issue?

**Comment by Piet-G** [comment]
We're seemingly having the same or a similar issue:

 On some devices permissions start disabled and the camera/microphone buttons don't show a popup to enable them thus making the buttons nonfunctional without feedback
 On some devices (e.g Pixel 4A Android 13) the permissions are initially asked, but if you disable them then the buttons are forever non functional without popup
 On some devices (e.g., Oneplus Nord, Android 12) the permissions popup consistently works

**Comment by Marabushka** [comment]
@saghul , same problem. 
Permission to access the microphone accepted, but the microphone button does not respond, when the application is restarted it starts to work
Version: 10
Platform: Android

**Comment by saghul** [comment]
Please test SDK 10.3.0 and post adb logcat logs.

**Comment by alexander-potemkin** [comment]
> Please test SDK 10.3.0 and post adb logcat logs.

I'm getting this problem regularly, just got it today on fresh Android and freshy downloaded Jitsi app. As the issue is quite complicated to reproduce (to keep adb connected) and it exists on both iOS & Android - is there probably - some way to setup some debug version of Jitsi, that will display logs / errors, when the issue re-occurs, to collect that information that way finally?

**Comment by viking2917** [comment]
I have more or less the same issue, but recorded in a different issue. 

https://github.com/issues/created?issue=jitsi%7Cjitsi-meet%7C16191

There is an adb log there, along with a possible? cause, related to included permissions...

**Comment by alexander-potemkin** [comment]
> I have more or less the same issue, but recorded in a different issue.
> 
> [#16191](https://github.com/jitsi/jitsi-meet/issues/16191)
> 
> There is an adb log there, along with a possible? cause, related to included permissions...

Thanks for sharing!!
